mirrord/agent/src/file.rs

```diff
@@ -727,6 +721,17 @@ impl FileManager {
         Ok(result)
     }
 
+    #[tracing::instrument(level = Level::DEBUG, skip(self), err)]
+    pub(crate) fn read_dir_batch(&mut self, fd: u64) -> RemoteResult<ReadDirBatchResponse> {
+        let result = self
+            .get_dir_stream(fd)?
+            .map(DirEntryInternal::try_from)
+            .try_collect::<Vec<_>>()
+            .map(|dir_entries| ReadDirBatchResponse { fd, dir_entries })?;
```

Comment:
Do we want to read the *whole* directory? A directory on ext4 can have like 64k entries

Reply:
+1, I'd expect some buffering

Reply:
Changed it! Now it's being done in steps. Should I add the `amount` we read as config somewhere?

Reply:
Nah, I think we can enable configuration later if at all